scripts/test-crds.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+# ------------------------------------------------------------------------------
+# test-crds.sh
+#
+# This script tests that our legacy v1beta1 versions of our CRDs upgrade cleanly
+# to their v1 versions, and that both the legacy and the v1 versions of the CRDs
+# both function with our controllers.
+#
+# NOTE: these tests are sensitive to which Kubernetes version they are run on.
+#       they will only run on Kubernetes clusters between v1.16.x and 1.21.x
+#       (inclusively) as v1.15.x is too old for the new CRDs and v1.22.x is to new
+#       for the old CRDs. At some point in the future (potentially when <v1.22.x is
+#       no longer supported by the chart) we can remove these tests.
+# ------------------------------------------------------------------------------
+
+# ------------------------------------------------------------------------------
+# Configuration
+# ------------------------------------------------------------------------------
+
+RELEASE_NAME="${RELEASE_NAME:-chart-tests-crds}"
+RELEASE_NAMESPACE="${RELEASE_NAMESPACE:-$(uuidgen)}"
+TEST_ENV_NAME="${TEST_ENV_NAME:-kong-charts-tests}"
+
+set -euo pipefail
+
+kubectl cluster-info --context kind-"${TEST_ENV_NAME}" 1>/dev/null
+KUBERNETES_VERSION="$(kubectl version -o json | jq -r '.serverVersion.gitVersion')"
+
+echo "INFO: running CRD tests on Kubernetes ${KUBERNETES_VERSION}"
+
+# ------------------------------------------------------------------------------
+# Cleanup Existing CRDs and Deploy Legacy CRDs
+# ------------------------------------------------------------------------------
+
+echo "INFO: cleaning up existing CRDs from the test cluster"
+for CRD in $(kubectl get crds -o=go-template='{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' |grep 'konghq.com')
+do
+    kubectl delete crd "${CRD}"
+done
+
+LAST_CRD_UPDATE="d2442de89cc036843c5f552c7daa6b023405f365"
+kubectl apply -f "https://raw.githubusercontent.com/Kong/kubernetes-ingress-controller/${LAST_CRD_UPDATE}/deploy/manifests/base/custom-types.yaml"
+
+# ------------------------------------------------------------------------------
+# Deploy Chart - Kubernetes Ingress Controller
+# ------------------------------------------------------------------------------
+
+cd charts/kong/
```

Comment:
implicit assumption that this script will be run with CWD set to the root of the repo. 

1. Please consider using this pattern to ensure the CWD at the beginning of the script: https://stackoverflow.com/a/59916
2. Instead of `cd`, consider replacing the `./` in the `helm install` invocation with a `./charts/kong/` to avoid unnecessarily altering the global state of the script with a `cd`.

Reply:
43af719e3fd00ffa0349a039829ba4e6defee73b

Reply:
and d63d5eb836629e7ccb1a9317944772fe76361774